src/utils/thread_helper.cpp

```diff
@@ -1360,8 +1392,48 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
         }
         // End of Nat64Mapping section.
 #endif // OTBR_ENABLE_NAT64
+#if OTBR_ENABLE_DHCP6_PD
+        // Start of Dhcp6PdState section.
+        {
+            wpanBorderRouter->set_dhcp6_pd_state(
+                Dhcp6PdStateFromOtDhcp6PdState(otBorderRoutingDhcp6PdGetState(mInstance)));
+        }
+        // End of Dhcp6PdState section.
 
-        // End of WpanBorderRouter section.
+        // Start of Hashed PD prefix
+        {
+            otBorderRoutingPrefixTableEntry aPrefixInfo;
+            Sha256::Hash                    hash;
+            Sha256                          sha256;
+
+            otBorderRoutingGetPdOmrPrefix(mInstance, &aPrefixInfo);
+            uint8_t pdPrefixShaInput[OT_IP6_ADDRESS_SIZE + kNat64PdCommonHashSaltLength];
+            memcpy(pdPrefixShaInput, aPrefixInfo.mPrefix.mPrefix.mFields.m8,
+                   sizeof(aPrefixInfo.mPrefix.mPrefix.mFields.m8));
```

Comment:
PD prefix will be shorter than 128 bits long, but here we are using the entire `OT_IP6_ADDRESS_SIZE`. Is this correct/desired?

Reply:
Updated according to [go/thread-ipaddress-obfuscation](https://goto.google.com/thread-ipaddress-obfuscation)